lib/vector-core/src/enrichment/tables.rs

```diff
@@ -123,12 +132,19 @@ impl vrl_core::EnrichmentTableSearch for TableSearch {
         &self,
         table: &str,
         criteria: Vec<vrl_core::Condition>,
-    ) -> Result<Option<Vec<String>>, String> {
+    ) -> Result<Option<BTreeMap<String, vrl_core::Value>>, String> {
         let tables = self.0.load();
         if let Some(ref tables) = **tables {
             match tables.get(table) {
                 None => Err(format!("table {} not loaded", table)),
-                Some(table) => Ok(table.find_table_row(criteria).cloned()),
+                Some(table) => Ok(table.find_table_row(criteria).map(|table| {
+                    table
+                        .iter()
+                        .map(|(key, value)| {
+                            (key.to_string(), vrl_core::Value::from(value.as_str()))
```

Comment:
```suggestion
                            (key.to_string(), value.into())
```

Not sure if this'll work?